### PR TITLE
first try

### PR DESCRIPTION
**Pull request by Ppllupold**

### CODE REVIEW

**Comment by Ppllupold** [comment]
i don`t know what`s the third commit, but i hope, nothing bad, sorry in advance

File: src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.dataconverter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int operationIndex = 0;
+    private static final int fruitIndex = 1;
+    private static final int amountIndex = 2;
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java:10]: please make your constant names uppercased

File: src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    public List<FruitTransaction> convert(List<String> csvLines) {
+        return csvLines.stream()
+                .skip(1)
+                .map(l -> l.split(","))
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java:16]: let's be consistent and introduce constant for ',' separator

File: src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                        arr[fruitIndex], Integer.parseInt(arr[amountIndex])))
+                .collect(Collectors.toList());
+    }
+
+    private FruitTransaction.Operation convertToOperation(String mark) {
+        for (FruitTransaction.Operation operation : FruitTransaction.Operation.values()) {
+            if (operation.getMarkInFile().equals(mark)) {
+                return operation;
+            }
+        }
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/dataconverter/DataConverterImpl.java:27]: let's move this logic to Operation enum. Create a method inside Operation that will return an operation based on string representation

File: src/main/java/core/basesyntax/model/FruitTransaction.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        return operation;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String markInFile;
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/model/FruitTransaction.java:32]: slightly confusing field name

File: src/main/java/core/basesyntax/reader/CsvFileReaderImpl.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @Override
+    public List<String> readFromCsv(String fileName) {
+        List<String> listOfLines = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                listOfLines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading the file: " + e);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/reader/CsvFileReaderImpl.java:19]: please specify the file name as well in the exception message

File: src/main/java/core/basesyntax/service/ReportGeneratorImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    private final ShopService shopService;
+
+    public ReportGeneratorImpl(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public String getReport() {
+        Map<String, Integer> storage = shopService.getStorage();
+        return "fruit,quantity\n"
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/service/ReportGeneratorImpl.java:16]: let's use System.lineseparator instead. Please fix in all places

File: src/main/java/core/basesyntax/service/ShopService.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public interface ShopService {
+    void process(List<FruitTransaction> fruitTransactions);
+
+    Map<String, Integer> getStorage();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/service/ShopService.java:10]: this is the redundant method. Your ShopService shouldn't be an entry point for storage retrieval. Storage is an independent entity and shouldn't be associated with ShopService

File: src/main/java/core/basesyntax/service/ShopServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operationhandlers.OperationHandler;
+import core.basesyntax.service.operationstrategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategyImpl operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/service/ShopServiceImpl.java:12]: as described above, let's move storage to the separate class

File: src/main/java/core/basesyntax/service/operationhandlers/PurchaseOperationHandler.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operationhandlers;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.merge(transaction.getFruit(),-transaction.getAmount(), Integer::sum);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/service/operationhandlers/PurchaseOperationHandler.java:9]: please fix style issues (indendations)

**Review by sarakhmen — CHANGES_REQUESTED** [review]
Overall good job, but let's improve your solution :) Also I don't see your main method/class. Could you please add it?

**Review by MAMentorFX — APPROVED** [review]

**Pull request closed without merging**